compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -143,8 +144,12 @@ class ClassfileParser(
       setClassInfo(moduleRoot, staticInfo)
     }
 
-    classRoot.hack = result
-    moduleRoot.hack = result
+    result match {
+      case result @ Some(_: DottyUnpickler) =>
+        classRoot.dottyUnpickler = result.asInstanceOf[Option[DottyUnpickler]]
```

Comment:
I see we have the same kind of problems :). https://github.com/dotty-staging/dotty/commit/b552b8353ef077991a0a5eb9eae11fb1cb8b8523